lib/engine/game/g_1848/step/dividend.rb

```diff
@@ -38,6 +38,13 @@ def process_dividend(action)
             revenue = calculate_boe_revenue
             payout = send(:payout, entity, revenue)
             payout_shares(entity, revenue) if payout[:per_share].positive?
+            action.id = @game.actions.last.id
```

Comment:
why is this line needed?

Reply:
`action` needs an 'id' or else it won't render BoE's revenue as a clickable hyperlink when stored in the operating history

The BoE revenue isn't a "real action" that's stored in the state, so we just set it to the last real action. 

I do the same thing here for the case when a company skips its revenue step on its first operating round
https://github.com/benjaminxscott/18xx/blame/57f9f3e26577523aa7ce77b3a2850c8f8f668adf/lib/engine/step/dividend.rb#L36